services/app-api/handlers/etl/numberFieldDataToFile.ts

```diff
@@ -1,215 +1,341 @@
+          report: enhanceMetadata(metadata, reportData),
+          fields: fieldsToFix,
+        });
+      }
 
-//iterate over the formTemplate & use that information to iterate over the fieldData
-export const extractNumericalData = (
-  formTemplate: AnyObject,
-  fieldData: AnyObject
-) => {
-  let numericFields: FieldTemplate[] = [];
-  iterateOverNumericFields(formTemplate.routes, numericFields);
+      console.log(`Processed report ${metadata.id} - ${performance.now()}`);
+    }
+  }
 
-  let extractedFieldData: FieldData[] = [];
+  console.log(
+    `${reportsToFix.length} reports need fixup: ${reportsToFix.map(
+      (rtf) => rtf.report.id
+    )}`
+  );
+  console.log(formatResultsAsTable(reportsToFix));
+  return reportsToFix;
+}
 
-  numericFields.forEach((item) => {
-    let fieldId = item.fieldId;
-    fieldValueById(fieldData, fieldId, extractedFieldData);
+/**
+ * Scan the entire form template version table,
+ * and fetch each template from the bucket.
+ */
+const getAllFormTemplates = async function* () {
+  const tableIterator = dynamodbLib.scanIterator({
+    TableName: formTemplateTableName,
   });
 
-  return extractedFieldData;
+  for await (const formTemplateMetadata of tableIterator) {
+    const reportJson = (await s3Lib.get({
+      Bucket: reportBuckets[formTemplateMetadata.reportType as ReportType],
+      Key: getFormTemplateKey(formTemplateMetadata.id),
+    })) as ReportJson;
+    yield {
+      id: formTemplateMetadata.id,
+      ...reportJson,
+    };
+  }
 };
 
-export const iterateOverNumericFields = (
-  formTemplateRoutes: any[],
-  store: any[]
-) => {
-  //check if the key is a number (array) or an object key matching the routeKeyList
-  let route: any[] = Object.entries(formTemplateRoutes).flatMap(
-    ([key, value]) => {
-      return routeKeyList.includes(key) || !isNaN(parseInt(key)) ? value : null;
+/**
+ * Given an array of report routes, iterate through them and all of their children
+ */
+const iterateRoutesIn = function* (
+  routes: ReportRoute[]
+): Generator<ReportRoute, void, undefined> {
+  for (let route of routes) {
+    yield route;
+    if (route.children) {
+      yield* iterateRoutesIn(route.children);
     }
-  );
+  }
+};
 
-  for (let arr of route) {
-    if (arr != null) {
-      if (
-        (arr.validation && arr.validation === "number") ||
-        (arr.type && arr.type === "number")
-      )
-        store.push({ fieldId: arr.id });
+/**
+ * Given a report route, iterate through all of its forms
+ */
+const iterateFormsIn = (route: ReportRoute) => {
+  // This filter catches form, modalForm, drawerForm, and overlayForm
+  return Object.values(route).filter((val: any): val is FormJson => val.fields);
+};
 
-      Object.keys(arr).forEach((key) => {
-        if (typeof arr[key] === "object") {
-          iterateOverNumericFields(arr[key], store);
+/**
+ * Given an array of form fields, iterate through them and all of their children
+ */
+const iterateFieldsIn = function* (
```

Comment:
No. Iterating through a collection of form fields is a complex endeavor. It's necessary for this script, it can't be made any simpler, and breaking it up into smaller functions will actually make it _harder_ to follow.